mlflow/models/utils.py

```diff
@@ -884,3 +887,58 @@ def get_model_version_from_model_uri(model_uri):
     (name, version) = get_model_name_and_version(client, model_uri)
     model_version = client.get_model_version(name, version)
     return model_version
+
+
+def _enforce_params_schema(params: Optional[Dict[str, Any]], schema: Optional[ParamSchema]):
+    if schema is None:
+        if params in [None, {}]:
+            return params
+        raise MlflowException(
```

Comment:
Can we use the `INVALID_PARAMETER_VALUE` error code for all exceptions thrown when validating parameters? Currently, the error code returned is 500 (internal error), which would cause clients to retry with exponential backoff

Can we add / update some test cases to assert that the exception error code is correct?

Example 500 response (should be 400:
```
<Response [500]>
{"error_code": "INTERNAL_ERROR", "message": "Parameters schema must be provided if `params` are provided.Please double check your model signature and ensure that params schema is provided."}
```